src/Sheet/index.jsx

```diff
@@ -20,13 +20,40 @@ export const VARIANTS = {
 class Sheet extends React.Component {
   constructor(props) {
     super(props);
+
+    this.wrapperRef = React.createRef();
     this.renderSheet = this.renderSheet.bind(this);
+    this.handleEscape = this.handleEscape.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleEscape);
+    document.addEventListener('mousedown', this.handleClickOutside);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleEscape);
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  }
+
+  handleEscape(e) {
+    if (e.key === 'Escape' && !this.props.blocking) {
+      this.props.onClose();
+    }
+  }
+
+  handleClickOutside(e) {
+    if (this.wrapperRef?.current && !this.wrapperRef.current.contains(e.target) && !this.props.blocking) {
+      this.props.onClose();
+    }
   }
```

Comment:
[curious] I wonder if we could use the callback functions from `FocusOn` here to simplify this?

<img width="582" alt="image" src="https://user-images.githubusercontent.com/2828721/162437341-2d7aefd9-c312-430e-a92e-3f90e81a57f3.png">

Reply:
the only thing, I didn't manage how to test these new events, so I removed new tests